src/macros/macro.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <QDebug>
+#include <QtCore>
+
+namespace {
+constexpr size_t kMaxMacroSize = 1000;
+const QLoggingCategory macros("macros");
+} // namespace
+
+/// A MacroAction is the smallest piece of a Macro.
+/// It contains a position as well as the action to be taken at that position
+/// (currently only jumps to a target position are available).
+struct MacroAction {
+    MacroAction(){};
+    MacroAction(double position, double target)
+            : position(position), target(target){};
+    /// TODO(xerus) use FramePos once https://github.com/mixxxdj/mixxx/pull/2861 is merged
+    double position;
+    double target;
+};
+
+/// A Macro stores a list of actions.
+/// The action list is pre-populated at construction so that it can be written to in real-time code.
+class Macro {
+  public:
+    MacroAction actions[kMaxMacroSize];
+
+    Macro() {
+        std::fill_n(actions, kMaxMacroSize, MacroAction());
+    }
+
+    /// Append a jump action to this Macro by assigning the next available slot.
+    /// Only called from RT.
+    void appendJump(double origin, double target);
+
+    /// The amount of saved Actions
```

Comment:
```suggestion
    /// The amount of saved MacroActions
```

Reply:
Isn't that implicit by it being in the Macro?
And while we're at it, I considered nesting the Action under the Macro class, is that considered better or worse style?

Reply:
> Isn't that implicit by it being in the Macro?

Yes, but it's easier to read code that is explicit. Also some tooling might recognize that MacroAction matches the name of a class.

>  And while we're at it, I considered nesting the Action under the Macro class, is that considered better or worse style?

I think it's fine how it is.